src/server/main_service.cc

```diff
@@ -2161,6 +2161,18 @@ void Service::PubsubPatterns(ConnectionContext* cntx) {
   (*cntx)->SendLong(pattern_count);
 }
 
+void Service::PubsubNumSub(const vector<string_view>& channels, ConnectionContext* cntx) {
+  vector<std::pair<std::string, long>> pubsub_num_array;
+  pubsub_num_array.reserve(channels.size() * 2);
```

Comment:
Why reserve `size()*2` instead of just `size()`?
(it's a `vector` of `pair`s so each element has both values)

Reply:
fixed: https://github.com/dragonflydb/dragonfly/pull/2282/commits/dd163a0a73f229a253443d525d8367d588aafab4